content/addons.md

```diff
@@ -138,20 +146,216 @@ There are several options to see the addon in action. We could use `npm link` or
+```css
+@import 'our-addon-name.css'
+/* The app's own app/styles/app.css */
+```
+
+If there are any problems getting this to work, one strategy is to build the addon with `ember build` and look inside the `dist` folder. The `dist` folder may be hidden by default in some code editors. The `dist` folder gives clues about what the consuming app sees as the file structure of the addon. See if the stylesheets are in `dist/assets/`. Then, in the Ember app, run `ember build` and look in the `dist` folder. We should see our stylesheets in `dist/assets` of the app too.
+
+#### Using CSS preprocessors for the addon's stylesheets
+
+While this guide focuses on the "out of the box" behavior of addons and the Ember CLI, there are some well-established patterns for handling stylesheets in a way that is scalable and maintainable. A CSS preprocessor like Sass (aka SCSS) allows you to nest style rules, use variables, and do simple mathematical operations.
+
+The best way to learn how to CSS preprocessors in your addon is to consult the documentation for the preprocessor addon of your choice, and study how other addon authors have implemented stylesheets. For example, [ember-styleguide](https://github.com/ember-learn/ember-styleguide/) is a UI component library that was made for the main Ember websites. It uses [ember-cli-sass](https://www.emberobserver.com/addons/ember-cli-sass) to manage styles. You can search [Ember Observer](https://emberobserver.com) for many more examples of styling in action!
+
+### Adding UI functionality with JavaScript
+
+There are two main types of JavaScript functionality that an addon can provide: API methods that developers can use after importing your addon, and interactivity JavaScript that is part of UI components. We'll cover UI use cases first.
+
+Interactivity in an addon can be handled the same way that it is done for an Ember app component. Every addon component template has a corresponding JavaScript file to go with it. For example, an addon can have its own actions, computed properties, and imported dependencies. Developers using the addon can pass in their own actions and attributes.
+
+For more information about building interactivity for your addon, reference the [Ember Guides components section](https://guides.emberjs.com/release/components/defining-a-component/). Remember to test as you work!
+
+### Providing multiple templates in one addon
+
 
-### Block form component template
+## Writing a JavaScript utilties addon
 
-### Adding JavaScript functionality
+Many addons have no UI components in them, or they offer a combination of JavaScript utilities and template helpers. In the regular npm ecosystem, JavaScript utility libraries are some of the most common packages. Although we could write a normal node package, providing an Ember addon to developers has some advantages. The developers don't need to worry about how to import a normal npm package. They can use `ember install our-addon-name` and get going right away.
```

Comment:
Another reason to create an addon vs a general package would be needing to use ember or ember-cli specific APIs.